modbus4mqtt/modbus4mqtt.py

```diff
@@ -65,6 +71,8 @@ def connect_mqtt(self):
         self._mqtt_client._on_disconnect = self._on_disconnect
         self._mqtt_client._on_message = self._on_message
         self._mqtt_client._on_subscribe = self._on_subscribe
+        if self.insecure is False:
```

Comment:
I'd prefer to gate the `tls_set()` call based on whether `self.cert is not None`, and set insecure mode using paho's `self._mqtt_client.tls_insecure_set(self.insecure)` instead.

Reply:
To access a server with valid (not self-signed) certificates, you don't need to pass a cert. I agree; the use of insecure is wrong. I need another param to turn on tls_set as paho won't auto try it.

Reply:
Sounds good to me!